runtime/doc/ui.txt

```diff
@@ -636,6 +636,13 @@ tabs.
 	purpose it only counts "virtual" or "displayed" lines, so folds
 	only count as one line.
 
+	All updates, such as `grid_line`, in a batch affects the new viewport,
+	despite the fact that `win_viewport` is received after the updates.
+	Applications implementing, for example, smooth scrolling should take
+	this into account and keep the grid separated from what's displayed on
+	the screen and copy it to the viewport destination once `win_viewport`
+	is received.
```

Comment:
Is the `win_viewport` still sent as part of the _same_ batch (logical batch, i e before the final "flush" event)? Then perhaps this specific callout is redundant or should be toned down a little. State between different objects is not fully synchronized until "flush" is received, this is just how all screen events work.

If the concern is _mainly_ the space-bar heating change of changing the order to what some existing UIs might have come expect, then it is better to call out such a change in `news.txt` rather than cluttering the docs.

Reply:
The documentation is the first place I go looking for this kind of information, and it also concerns people starting to make new UIs, or trying to fix bugs with their old information.

This would not be needed, if we somehow can change the order so that `win_viewport` comes before the `grid_scroll` and `grid_line` calls in the batch. But as it is now, coming the last in the batch, applications are forced to do some kind of double buffering.

Intuitively, users will think that things are sent in the correct order, and that flush is just a point where everything is consistent.